shared/slices/slice_generic.go

```diff
@@ -0,0 +1,72 @@
+package slices
+
+// Type is the placeholder type that indicates a generic value.
+// when executed variables of this type will be replaced with,
+// references to the specific types.
+type Type interface{}
+
+// GenericItem is of type empty interface in order
+// to hold any values.
+type GenericItem Type
```

Comment:
Do we need this at all or can the argument type/return type be `interface{}`?

Reply:
interface{} alone will work , will change it. Anyways I am figuring out the solution to support data type @nisdas  suggested.

Reply:
@prestonvanloon this removed